hunting/aws/docs/secretsmanager_high_frequency_get_secret_value.md

```diff
@@ -0,0 +1,42 @@
+# Secrets Manager High Frequency of Programmatic GetSecretValue API Calls
+
+---
+
+## Metadata
+
+- **Author:** Elastic
+- **Description:** This hunting query identifies when a high frequency of `GetSecretValue` API calls are made to the AWS Secrets Manager service programmatically. The `GetSecretValue` API call retrieves the secret value for a specified secret. High frequency of these calls may indicate an adversary attempting to access sensitive information stored in AWS Secrets Manager via a compromised account or automated tooling.
+
+- **UUID:** `ef244ca0-5e32-11ef-a8d3-f661ea17fbce`
+- **Integration:** [aws.cloudtrail](https://docs.elastic.co/integrations/aws/cloudtrail)
+- **Language:** `[ES|QL]`
+- **Source File:** [Secrets Manager High Frequency of Programmatic GetSecretValue API Calls](../queries/secretsmanager_high_frequency_get_secret_value.toml)
+
+## Query
+
+```sql
+from logs-aws.cloudtrail*
+| where @timestamp > now() - 7 day
+| where
+    event.provider == "secretsmanager.amazonaws.com"
+    and event.action == "GetSecretValue"
+    and user_agent.name not in ("Chrome","Firefox","Safari", "Edge", "Brave", "Opera")
+| dissect aws.cloudtrail.request_parameters "%{}secret:%{secret_value}}"
+| stats request_counts = count(*) by event.action, aws.cloudtrail.user_identity.arn, source.ip
```

Comment:
maybe adding to the `by` aggregation fields `user_agent.name` and filtering after that with a `where` condition count is less than a threshold (to spot rare user.agent)

Reply:
Definitely adding user agent would be nice. But limiting this to rare user-agents may miss the common valid account compromise. I added user agent and then sorted by ascending and added a note about rare user agents as an option to pivot on.